Dockerfile

```diff
@@ -0,0 +1,39 @@
+
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+ARG py_version=3.8
+FROM apache/beam_python${py_version}_sdk:2.40.0 as beam_sdk
+FROM continuumio/miniconda3:4.12.0
+
+# Update miniconda
+RUN conda update conda -y
+
+# Create conda env using environment.yml
+ARG weather_tools_git_rev=main
+RUN git clone https://github.com/google/weather-tools.git /weather
+WORKDIR /weather
+RUN git checkout "${weather_tools_git_rev}"
+RUN conda env create -f environment.yml
```

Comment:
I'm curious why we should clone the repo in the image vs. mapping the repo from the host machine into the Docker image. What's the advantage to doing it this way? If we did the latter, could this be simpler?

Reply:
That was how I originally wrote it, but the idea here is that `COPY . /weather` would cause any changes to the directory to undo all caching (i.e. the directory hash will change so it'll have to do the `conda env create` which is the slowest step from scratch)

Reply:
This is also more consistent with how the Conda recipe is built, as it will read from a specific branch/tag

Reply:
One alternative is just copying `weather_*` directories and `setup.py`, but it's not super elegant and makes it easier to break with changes in code structure

Reply:
Thanks for the explanation. This makes sense, and am happy with the existing implementation.